### DISCUSSION TITLE
How to get average power consumption for a specific hour of the day

### DISCUSSION THREAD

**Discussion by joao-l-cruz** [wlcrs/huawei_solar#542] [Setting up the integration]
Hi everyone,

I'm new at Homeassistant, jinja2 and yaml. I need help ! :-)

I'm trying to figure a way of getting the average power consumption for a specific hour of the day (average of the last X days)

The ideia would be to build a sensor able to give me that for the 24 hours of the day.

The simple question ... where to start ? I have no ideia ...

Thank you all.

**Comment by SvenAbels** [comment]
Hi,

there is the statistic diagramm card included in home assistant where you can select a bar chart type and the average as a type. Is't this what you mean?

**Reply by joao-l-cruz** [reply]
Hi Sven

Thank you for your reply. It's no the card I'm looking for but to be able to use the value on a template to do some calculations.


Thank you.

**Comment by JensenNick** [comment]
Looking for this too. Not a card or chart, but a sensor which returns the hourly mean value for a given sensor. I've got to the point where it is either an sql-lookup in your database or the statistics function in HA.

https://www.home-assistant.io/integrations/statistics/

**Reply by joao-l-cruz** [reply]
Hi Jensen,

sounds an excelent starting point, thank you !

Do you know how to build a template sensor with atributes where each attribute would be an hour of the day?

**Reply by JensenNick** [reply]
Not yet ;-)…

**Reply by joao-l-cruz** [reply]
Same here 🤪
